### PR TITLE
feat: display human-friendly error messages when Prisma's libraries cannot be initialized

### PR DESCRIPTION
**Pull request by jkomyno**
Contributes to https://github.com/prisma/prisma/issues/16970

### CODE REVIEW

File: packages/engine-core/src/common/errors/utils/handleEngineLoadingErrors.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+   * Prisma has detected an incompatible version of the \`glibc\` C standard library installed in your system. This probably means your system may be too old to run Prisma. Please refer to the documentation about Prisma's system requirements: https://pris.ly/d/system-requirements.
+   *
+   * Details: symbol __cxa_thread_atexit_impl, version GLIBC_2.18 not defined in file libc.so.6 with link time reference
+   *      at load (/usr/src/app/node_modules/prisma/build/index.js:93185:11)
+   *      at getEngineVersion (/usr/src/app/node_modules/prisma/build/index.js:93485:16)
+   *      at runMicrotasks (<anonymous>)
+   *      at processTicksAndRejections (internal/process/task_queues.js:95:5)
+   */
+  return `${errorTitle}
+${potentialReasonMessage}.
```
Comment (aqrln) [inline packages/engine-core/src/common/errors/utils/handleEngineLoadingErrors.ts:73]: Some `potentialReasonMessage`s end with a period, and some don't, which causes some of them to have two periods in the final error message: ``Prisma cannot find the required `libssl` system library in your system. Please install openssl and try again..``

Comment (jkomyno) [reply packages/engine-core/src/common/errors/utils/handleEngineLoadingErrors.ts:73]: Thanks, I had fixed it but apparently didn't push the file

**Comment by jkomyno** [comment]
Note: we had tried using `jest.spyOn` to test this errors in `LibraryEngine.test.ts`, with little success.
Here a snippet for us to try again in the future:

```typescript
import * as defaultLibraryLoaderUtils from '../library/DefaultLibraryLoader'

describe('PrismaClientInitializationError', () => {
  const engineConfig = {
    datamodelPath: '/mock',
    logEmitter: new EventEmitter(),
    tracingConfig: {
      enabled: false,
      middleware: false,
    },
    env: {},
  }
  test('symbol not found', async () => {
    expect.assertions(2)
    const spyLoadTag = `Unable to require(\`/app/node_modules/.pnpm/prisma@x.x.x/node_modules/prisma/libquery_engine-linux-arm64-openssl-1.1.x.so.node\`)
Error relocating /app/node_modules/.pnpm/prisma@x.x.x/node_modules/prisma/libquery_engine-linux-arm64-openssl-1.1.x.so.node: __res_init: symbol not found`
    jest.mock('../library/DefaultLibraryLoader', () => {
      return {
        __esModule: true,
        ...jest.requireActual('../library/DefaultLibraryLoader'),
      }
    })
    const loadMock = jest.spyOn(defaultLibraryLoaderUtils, 'load').mockImplementation((id: string) => {
      throw new Error(spyLoadTag)
    })
    const loader = new defaultLibraryLoaderUtils.DefaultLibraryLoader(engineConfig)
    try {
      await loader.loadLibrary()
    } catch (e) {
      expect(e).toBeInstanceOf(PrismaClientInitializationError)
      expect(e.message).toMatchInlineSnapshot()
    } finally {
      loadMock.mockRestore()
    }
  })
})
```

File: packages/engine-core/src/common/errors/utils/handleEngineLoadingErrors.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    )
+    .otherwise(() => {
+      return `The Prisma engines do not seem to be compatible with your system. ${referToSystemRequirementsDocs}.`
+    })
+
+  /**
+   * Example:
+   *
+   * Error: Unable to require(`/usr/src/app/node_modules/@prisma/engines/libquery_engine-linux-arm64-openssl-1.0.x.so.node`)
+   * Prisma has detected an incompatible version of the \`glibc\` C standard library installed in your system. This probably means your system may be too old to run Prisma. Please refer to the documentation about Prisma's system requirements: https://pris.ly/d/system-requirements.
```
Comment (aqrln) [inline packages/engine-core/src/common/errors/utils/handleEngineLoadingErrors.ts:64]: The period after the link may be treated as a part of the link by the terminal, or inadvertently copied manually. I’d probably remove it for messages that end with a link for practical reasons, and only keep it for normal sentences.

**Review by aqrln — APPROVED** [review]
Thank you!

**Pull request merged**